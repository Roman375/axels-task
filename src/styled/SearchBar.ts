import styled from 'styled-components'

export const Search = styled.form `
  display: flex;
  justify-content: center;
  width: 50%;
`
