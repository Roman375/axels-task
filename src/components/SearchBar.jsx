import React from 'react'

import { Search } from '../styled/SearchBar'

const SearchBar = ({handleChange}) => {
  return (
    <Search>
      <input
        className="form-control me-2"
        type="search"
        placeholder="Search"
        aria-label="Search"
        onChange={handleChange}
      />
    </Search>
  )
}

export default SearchBar
