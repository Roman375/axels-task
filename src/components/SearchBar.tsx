import React, { FC } from 'react'

import { eventType } from './UsersList'
import { Search } from '../styled/SearchBar'

type Props = {
  handleChange: (event: eventType) => void
}

const SearchBar: FC<Props> = ({ handleChange }) => (
  <Search>
    <input
      className="form-control me-2"
      type="search"
      placeholder="Search"
      aria-label="Search"
      onChange={handleChange}
    />
  </Search>
)

export default SearchBar
