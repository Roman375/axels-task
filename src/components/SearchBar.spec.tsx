import renderer from 'react-test-renderer'
import SearchBar from './SearchBar'

describe('<SearchBar />', () => {
  const props = {
    handleChange: () => {},
  }

  describe('With Snapshot Testing', () => {
    it('Render component SearchBar"', () => {
      const component = renderer.create(<SearchBar {...props} />)
      expect(component).toMatchSnapshot()
    })
  })
})
