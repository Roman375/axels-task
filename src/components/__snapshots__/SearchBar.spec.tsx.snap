// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> With Snapshot Testing Render component SearchBar" 1`] = `
<form
  className="sc-bdfBwQ hCXJer"
>
  <input
    aria-label="Search"
    className="form-control me-2"
    onChange={[Function]}
    placeholder="Search"
    type="search"
  />
</form>
`;
